### PR TITLE
Price group: hiding plans according to the price group.

### PR DESCRIPTION
**Pull request by southp**
#### Changes proposed in this Pull Request
**Description update: May. 7th.**

*This is part of the Global Pricing Optimization project. More context can be found in p2-pau2Xa-1de.*

This PR implements the plan hiding logic based on the marketing price group, for hiding plans accordingly in the signup flow, the site creation flow, and viewing `/plans` as a logged-in user.

#### Dependency

* https://github.com/Automattic/wp-calypso/pull/41871
* #41670 (shipped)
* code-D43026
* code-D42721 (shipped)

#### Testing instructions

Please refer to the test section of code-D42721 to see how to enforce a user into either test pricing group.

1. Enforce a test account to the test pricing group 1.
    1. Go through the sign up flow and stop at the plans step.
    1. While there shouldn't be any visible difference, `getState().signup.dependencyStore` should include a `marketing_price_group` property, with value `price_2020_q2_test_1`.
    1. Stay logged-in, go through the site creation flow and stop at the plans step. 
    1. Stay logged-in, go to `/plans`.     
1. Enforce a test account to the test pricing group 2.
    1. Go through the sign up flow and stop at the plans step.
    1. The Personal plan should be hidden.
    1. Stay logged-in, go through the site creation flow and stop at the plans step. The Personal plan should be hidden.
    1. Stay logged-in, go to `/plans`. The Personal plan should be hidden.
1. Enforce a test account to the test pricing group 3.
    1. Go through the sign up flow and stop at the plans step.
    1. The Personal plan and the Premium plan should be hidden.
    1. Stay logged-in, go through the site creation flow and stop at the plans step. The Personal plan and the Premium plan should be hidden.
    1. Stay logged-in, go to `/plans`. The Personal plan and the Premium plan should be hidden.

### CODE REVIEW

File: client/state/selectors/get-current-user-marketing-price-group.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Returns the marketing price group of the current user.
+ *
+ * @param {object} state Global state tree
+ * @returns {string?} The price group slug
+ */
+export default ( state ) => {
+	const currentUser = getCurrentUser( state );
+
+	return ( currentUser && currentUser.meta.marketing_price_group ) || null;
```
Comment (niranjan-uma-shankar) [inline client/state/selectors/get-current-user-marketing-price-group.js:15]: Small thing, but we could use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here:

> return currentUser?.meta.marketing_price_group
or the following if you want `null` returned instead of `undefined`
> return currentUser?.meta.marketing_price_group || null

**Review by niranjan-uma-shankar — COMMENTED** [review]
Looking good @southp. Found a minor issue.
I tested with the backend patch. When I visit the plans page just after signup, or right after "Add new site" flow is complete, then the plans page alignment is not proper. If I reload the page, then the alignment fixes itself. I would guess this is not a blocker, but just bringing to your notice. 
Check screen recording here - https://d.pr/v/JlTXPX 

Also, could you please rebase this branch so that I can test the changes done in https://github.com/Automattic/wp-calypso/pull/41526?

**Review by niranjan-uma-shankar — COMMENTED** [review]
I tested a bit more and I found a few places where we explicitly mention the Personal and Premium plan:

Screenshot 2020-05-04 at 6 28 05 PM

Screenshot 2020-05-04 at 6 29 05 PM

Screenshot 2020-05-04 at 7 03 53 PM

Screenshot 2020-05-04 at 7 02 05 PM

Screenshot 2020-05-04 at 6 57 16 PM

There are several more places where we show these upgrade nudges. I suggest the following search terms to find references in code:

https://github.com/Automattic/wp-calypso/search?q=%22WordPress.com+Premium%22&unscoped_q=%22WordPress.com+Premium%22

https://github.com/Automattic/wp-calypso/search?q=%22WordPress.com+Personal%22&unscoped_q=%22WordPress.com+Personal%22

We also refer to these plans in other ways, e.g. Premium plan, WordPress.com Premium, Upgrade to Premium etc. so we need to find all those references. For eg. users from Canada and US will see the following card in the Marketing page:

Screenshot 2020-05-04 at 7 22 20 PM

Some other uses of Premium/Personal variations:

Screenshot 2020-05-04 at 7 15 37 PM

Screenshot 2020-05-04 at 7 11 45 PM

**Comment by travisw** [comment]
> I tested with the backend patch. When I visit the plans page just after signup, or right after "Add new site" flow is complete, then the plans page alignment is not proper. If I reload the page, then the alignment fixes itself. I would guess this is not a blocker, but just bringing to your notice.

I also noticed something similar on the /plans page. Although a refresh doesn't seem to fix it for me. It doesn't look like the new CSS is loading for me for whatever reason.

**Comment by travisw** [comment]
> I tested a bit more and I found a few places where we explicitly mention the Personal and Premium plan:

Thanks for finding these @niranjan-uma-shankar - I mentioned in the P2 for more visibility as things to be aware of for the test. pbBQWj-7J-p2

File: client/state/selectors/get-current-user-marketing-price-group.js (hunk 1 above)
Comment (southp) [reply client/state/selectors/get-current-user-marketing-price-group.js]: Thanks for the tip! I have updated the code in fa49661.

**Comment by southp** [comment]
@niranjan-uma-shankar Now that all the dependent branches and the backend diff have been deployed, this PR should be ready for a revisit again. We aim at shipping this test within this week, so it would be wonderful if you can drive by when you have some time today :)

File: client/lib/signup/step-actions/index.js [hunk 2]
```diff
@@ -450,7 +450,10 @@ export function createAccount(
 				// Fire after a new user registers.
 				recordRegistration( { userData: registrationUserData, flow: flowName, type: 'social' } );
 
-				callback( undefined, pick( response, [ 'username', 'bearer_token' ] ) );
+				callback(
+					undefined,
+					pick( response, [ 'username', 'bearer_token', 'marketing_price_group' ] )
```
Comment (niranjan-uma-shankar) [inline client/lib/signup/step-actions/index.js:455]: Does the usersSocialNew endpoint return the marketing_price_group value? Also, like done in the else block, it's safer to assign to a const variable:

> const marketing_price_group = ( response && response.marketing_price_group ) || '';

**Comment by niranjan-uma-shankar** [comment]
@southp I tested and this works well. Left a comment on social signup above, but otherwise looking good. I can do a final run tomorrow after the social signup thing is fixed.

File: client/lib/signup/step-actions/index.js (hunk 2 above)
Comment (simison) [reply client/lib/signup/step-actions/index.js:455]: You can simplify tad bit to:
`response?.marketing_price_group ?? ''`

Comment (niranjan-uma-shankar) [reply client/lib/signup/step-actions/index.js:455]: Yes, optional chaining for the win. Looks cleaner.

Comment (southp) [reply client/lib/signup/step-actions/index.js]: Cool tip! Added in d48c50b.

**Comment by southp** [comment]
@niranjan-uma-shankar Thanks for finding it. It's definitely something I overlooked. The patch to update the social signup endpoint is opened as code-D43026.

Also, to address the code-wise issue, I refactored `createAccount()` in https://github.com/Automattic/wp-calypso/pull/41871 and then implemented the requested change here again.

**Review by niranjan-uma-shankar — COMMENTED** [review]
Do we want to add a primary button(pink colour) for the plan options in CAD and AUD? 

In signup flow:

Screenshot 2020-05-07 at 10 36 33 AM

In /plans page:

Screenshot 2020-05-07 at 10 37 28 AM

**Comment by southp** [comment]
@niranjan-uma-shankar Good catch. Since this PR and the related diff is already quite involving, I think it would be better to be fixed in a separate PR. Does that make sense to you?

**Comment by niranjan-uma-shankar** [comment]
Makes sense

**Review by niranjan-uma-shankar — COMMENTED** [review]
Noting here an issue which is an edge case, and I don't think we should try fixing it for this test. This is just for the record so that if we get that "lucky" user who faces the issue, we'll know the cause.

As you know, we have a domain-plan-swap test running but that's only for EN-US. We get  geo location from wpcom cookie. The cookie value stays even if the user moves to a different location i.e its value is not overwritten most of the times, except in some edge cases. Now, if a user satisfies the following conditions, there's a scenario in which they'll see an empty plans page:
1. User creates their account from Canada or Australia, and gets assigned to the pricing test. So, Personal & Premium plan is hidden.
2. The user later deletes their browser cookies for wpcom. Several ways why this could happen - either they flushed their browser, or used an incognito window for wpcom so cookies aren't stored, A more common cause could be that they later use a different device from the one they used to create the account.
3. The user logs in to wpcom later, but this time from a US IP address. Either they use a new device like mobile or tab from a US IP, or they use the same device but with cookies previously cleared. The wpcom cookie will now store "US" as the location. Their currency will still be CAD or AUD since that doesn't change.
4. If they click "Add new site", they will be assigned to the swap test. Say they got assigned to the variant, so they see the steps swapped. 
5. If they select a free plan -> custom domain -> click Yes on the upsell offer -> they will see an empty plans page(we show a modified plans page for those who click Yes on the upsell offer page with just the Personal & Premium plan, but this will be empty since both plans are removed).

Screen recording - https://d.pr/v/RB6ATS

Again, a lot of things have to align for this and it is an edge case.

**Review by niranjan-uma-shankar — APPROVED** [review]
LGTM :shipit:
